README.md

```diff
@@ -48,298 +48,400 @@ The FWS Payments contract enables ERC20 token payment flows through "rails" - au
-
-This model gives flexibility: this can guarantee ongoing service payments while also handling special, immediate payments as needed—all with clear, enforced separation and safety for both parties.
+-   **Settle Regularly:** You must settle rails frequently. A rail's `lockupPeriod` is a measure of the risk you are willing to take. If you wait longer than the `lockupPeriod` to settle, you allow a client to build up a payment obligation that may not be fully covered by the lockup guarantee if they become insolvent.
+-   **Monitor Client Solvency:** Use the `getAccountInfoIfSettled` function to check if a client is funded. If their `fundedUntilEpoch` is approaching the current epoch, they are at risk.
+-   **Terminate Proactively:** If a client becomes insolvent or unresponsive, terminate the rail immediately. This is the **only way** to activate the safety hatch and ensure you can claim payment from the funds guaranteed by the streaming lockup.
 
 ## Core Functions
 
 ### Account Management
 
+Functions for managing user accounts, including depositing and withdrawing funds. These functions support both ERC20 tokens and the native network token ($FIL) by using `address(0)` as the token address.
+
 #### `deposit(address token, address to, uint256 amount)`
 
-Deposits tokens into a specified account.
+Deposits tokens into a specified account. This is the standard method for funding an account if not using permits. It intelligently handles fee-on-transfer tokens by calculating the actual amount received by the contract.
 
+- **When to use:** Use this for direct transfers from a wallet or another contract that has already approved the Payments contract to spend tokens.
+- **Native Token (FIL):** To deposit the native network token, use `address(0)` for the `token` parameter and send the corresponding amount in the transaction's `value`.
 - **Parameters**:
-  - `token`: ERC20 token contract address
-  - `to`: Recipient account address
-  - `amount`: Token amount to deposit
+  - `token`: ERC20 token contract address.
+  - `to`: The account address to credit with the deposit.
+  - `amount`: The amount of tokens to transfer.
 - **Requirements**:
-  - Caller must have approved the contract to transfer tokens
+  - For ERC20s, the direct caller (`msg.sender`) must have approved the Payments contract to transfer at least `amount` of the specified `token`.
+  - For the native token, `msg.value` must equal `amount`.
 
 #### `depositWithPermit(address token, address to, uint256 amount, uint256 deadline, uint8 v, bytes32 r, bytes32 s)`
 
-Deposits tokens using EIP-2612 permit.
+Deposits tokens using an EIP-2612 permit, allowing for gasless token approval.
 
+- **When to use:** Ideal for user-facing applications where the user can sign a permit off-chain. This combines approval and deposit into a single on-chain transaction, saving gas and improving user experience.
+- **Note:** This function is for ERC20 tokens only and does not support the native token.
```

Comment:
> does not support the native token
oh, this is good to know! I would have just assumed it would be smart enough to figure that out; should we just add it there since it should be a pretty easy test?